node/src/components/block_proposer.rs

```diff
@@ -409,73 +402,68 @@ impl BlockProposerReady {
         past_deploys: HashSet<DeployHash>,
         random_bit: bool,
     ) -> ProtoBlock {
-        let max_transfers = deploy_config.block_max_transfer_count as usize;
-        let max_deploys = deploy_config.block_max_deploy_count as usize;
-        let max_block_size_bytes = deploy_config.max_block_size as usize;
-        let block_gas_limit = Gas::from(deploy_config.block_gas_limit);
-
-        let mut transfer_hashes = Vec::new();
-        let mut deploy_hashes = Vec::new();
-        let mut block_gas_running_total = Gas::zero();
-        let mut block_size_running_total = 0usize;
-
-        for (hash, deploy_type) in self.sets.pending.iter() {
-            let at_max_transfers = transfer_hashes.len() == max_transfers;
-            let at_max_deploys = deploy_hashes.len() == max_deploys
-                || (deploy_type.is_wasm()
-                    && block_size_running_total + DEPLOY_APPROX_MIN_SIZE >= max_block_size_bytes);
-
-            if at_max_deploys && at_max_transfers {
-                break;
+        let mut appendable_block = AppendableBlock::new(deploy_config, block_timestamp);
+
+        // We prioritize transfers over deploys, so we try to include them first.
+        for (hash, deploy_type) in &self.sets.pending {
```

Comment:
I think it's a good idea to separate Wasm from native transfers in the pending sets.

Reply:
I agree, but that may also be a separate task, for after the next release.